src/main/java/github/scarsz/discordsrv/DiscordSRV.java

```diff
@@ -721,19 +726,42 @@ private List<InetAddress> lookupPublic(String host) throws UnknownHostException
         dispatcher.setMaxRequestsPerHost(20); // most requests are to discord.com
         ConnectionPool connectionPool = new ConnectionPool(5, 10, TimeUnit.SECONDS);
 
-        OkHttpClient httpClient = new OkHttpClient.Builder()
+        String proxyHost = config.getString("ProxyHost").trim();
+        String proxyPort = config.getString("ProxyPort").trim();
+        final String authUser = config.getString("ProxyUser").trim();
+        final String authPassword = config.getString("ProxyPassword").trim();
+
+        OkHttpClient httpClient; // Initialized below
+        OkHttpClient.Builder httpClientBuilder = new OkHttpClient.Builder()
                 .dispatcher(dispatcher)
                 .connectionPool(connectionPool)
                 .dns(dns)
-                // more lenient timeouts (normally 10 seconds for these 3)
                 .connectTimeout(20, TimeUnit.SECONDS)
                 .readTimeout(20, TimeUnit.SECONDS)
                 .writeTimeout(20, TimeUnit.SECONDS)
                 .hostnameVerifier(noopHostnameVerifier.isPresent() && noopHostnameVerifier.get()
                         ? (hostname, sslSession) -> true
-                        : OkHostnameVerifier.INSTANCE
-                )
-                .build();
+                        : OkHostnameVerifier.INSTANCE);
+
+        // Reduce proxy username and password logic duplication
+        if (proxyHost != null && !proxyHost.isEmpty()) {
+            // This had to be set to empty string to avoid issue with basic auth
+            System.setProperty("jdk.http.auth.tunneling.disabledSchemes", "");
+            Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyHost, Integer.parseInt(proxyPort)));
+            httpClientBuilder = httpClientBuilder.proxy(proxy);
+
+            if (authPassword != null && !authPassword.isEmpty()) {
+                httpClientBuilder = httpClientBuilder.proxyAuthenticator(new Authenticator() {
+                    @Override
+                    public Request authenticate(Route route, Response response) throws IOException {
+                        String credential = Credentials.basic(authUser, authPassword);
+                        return response.request().newBuilder().header("Proxy-Authorization", credential).build();
+                    }
+                });
+            }
+        } 
+
+        httpClient = httpClientBuilder.build();
```

Comment:
```suggestion
        OkHttpClient httpClient = httpClientBuilder.build();
```
assign the variable on the same line where you initialize it.